index.js

```diff
@@ -0,0 +1,133 @@
+const express = require('express')
+const teams = require('./teams.json')
+
+const app = express()
+
+
+
+app.get('/teams', (request, response) => {
+
+    response.send(teams)
+
+})
+
+app.get('/teams/', (request, response) => {
+
+    const allTeams = teams.map((team) => {
+
+        return team
+
+    })
+})
+
+
+
+app.get('/teams/:id', (request, response) => {
+    let matchingTeam = teams.filter((team) => {
+        return team.id === parseInt(request.params.id)
+    })
+    let matchingAbb = teams.filter((team) => {
+        return team.abbreviation === (request.params.id.toUpperCase())
+    })
```

Comment:
Because you are doing both of these, you are looping twice for every request a user makes rather than having this is your `if(!isNaN())` check so you only do one or even better using the `OR` approach discussed in class.